lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -0,0 +1,413 @@
+        -- Jörmungandr, this is therefore useless to have in 'normal' BlockHeader
+        --
+        -- Yet, since we also serialize these to the database, this requires
+        -- some non-trivial changes. Not fixing this right now is also a
+        -- possibility.
+        , parentHeaderHash = W.Hash "parentHeaderHash - unused in Shelley"
+        }
+
+fromSlotLength :: SlotLength -> W.SlotLength
+fromSlotLength = W.SlotLength
+    . getSlotLength
+
+-- NOTE: Unsafe conversion from Natural -> Word16
+fromMaxTxSize :: Natural -> Quantity "byte" Word16
+fromMaxTxSize =
+    Quantity . fromIntegral
+
+fromPParams :: SL.PParams -> W.TxParameters
+fromPParams pp = W.TxParameters
+    { getFeePolicy = W.LinearFee
+        (Quantity (naturalToDouble $ SL._minfeeA pp))
+        (Quantity (fromIntegral $ SL._minfeeB pp))
+        (Quantity 0) -- TODO: it's not as simple as this?
+    , getTxMaxSize = fromMaxTxSize $ SL._maxTxSize pp
+    }
+  where
+    naturalToDouble :: Natural -> Double
+    naturalToDouble = naturalToDouble
+
+-- | Convert genesis data into blockchain params and an initial set of UTxO
+fromGenesisData
+    :: ShelleyGenesis TPraosStandardCrypto
+    -> (W.GenesisBlockParameters, [W.TxOut])
+fromGenesisData g =
+    ( W.GenesisBlockParameters
+        { staticParameters = W.BlockchainParameters
+            { getGenesisBlockHash =
+                -- TODO: There is not yet any agreed upon definition of a
+                -- genesis hash for a shelley-only testnet.
+                --
+                -- For now we use a dummy value.
+                W.Hash . BS.pack $ replicate 32 1
+            , getGenesisBlockDate =
+                W.StartTime . getSystemStart . sgStartTime $ g
+            , getSlotLength =
+                fromSlotLength . sgSlotLength $ g
+            , getEpochLength =
+                W.EpochLength . fromIntegral . unEpochSize . sgEpochLength $ g
+            , getEpochStability =
+                Quantity . fromIntegral . maxRollbacks . sgSecurityParam $ g
+            , getActiveSlotCoefficient =
+                W.ActiveSlotCoefficient 1.0
+            }
+        , txParameters =
+            -- TODO: implement properly when possible
+            -- From where are we supposed to get them?
+            W.TxParameters
+                { getFeePolicy =
+                    W.LinearFee (Quantity 0) (Quantity 0) (Quantity 0)
+                , getTxMaxSize =
+                    Quantity 1000
+                }
+        }
+    , map mkTxOut . Map.toList . sgInitialFunds $ g
+    )
+  where
+    mkTxOut (addr, coin) = W.TxOut
+        (W.Address $ serialize' addr)
+        (W.Coin $ fromIntegral coin)
+
+fromNetworkMagic :: NetworkMagic -> W.ProtocolMagic
+fromNetworkMagic (NetworkMagic magic) =
+    W.ProtocolMagic (fromIntegral magic)
+
+fromProtocolMagicId :: ProtocolMagicId -> W.ProtocolMagic
+fromProtocolMagicId = W.ProtocolMagic . fromIntegral . unProtocolMagicId
```

Comment:
I thought the concept of protocolMagic was gone in Shelley :thinking: ? This one is probably unused.

Reply:
Done